rpcperms/middleware_handler.go

```diff
@@ -404,12 +409,15 @@ func NewMessageInterceptionRequest(ctx context.Context,
 		return nil, err
 	}
 
+	md, _ := metadata.FromIncomingContext(ctx)
+
 	req := &InterceptionRequest{
-		Type:        authType,
-		StreamRPC:   isStream,
-		Macaroon:    mac,
-		RawMacaroon: rawMacaroon,
-		FullURI:     fullMethod,
+		Type:             authType,
+		StreamRPC:        isStream,
```

Comment:
Hmm, thinking about how to test this for a streaming RPC, I realized that metadata can only be passed once per stream, not for each individual message. Is that okay for the use case we have in mind?

Reply:
currently for the use case in lit, we dont need this for a stream call... 

But i think even if we did, it _should_ be ok since we can get what we need from the first message